test/unittests/execute_test.cpp

```diff
@@ -93,68 +93,58 @@ TEST(execute, local_tee)
 
 TEST(execute, global_get)
 {
-    Module module;
-    module.funcsec.emplace_back(TypeIdx{0});
-    module.globalsec.emplace_back(Global{true, {ConstantExpression::Kind::Constant, {42}}});
-    module.codesec.emplace_back(Code{0, {Instr::global_get, Instr::end}, {0, 0, 0, 0}});
-
-    auto instance = instantiate(module);
-
-    const auto [trap, ret] = execute(instance, 0, {});
-
-    ASSERT_FALSE(trap);
-    ASSERT_EQ(ret.size(), 1);
-    EXPECT_EQ(ret[0], 42);
+    /* wat2wasm
+    (global i32 (i32.const 42))
+    (func (result i32)
+      get_global 0
+    )
+    */
+    const auto wasm =
+        from_hex("0061736d010000000105016000017f030201000606017f00412a0b0a0601040023000b");
+    EXPECT_RESULT(execute(parse(wasm), 0, {}), 42);
 }
 
 TEST(execute, global_get_two_globals)
 {
-    Module module;
-    module.funcsec.emplace_back(TypeIdx{0});
-    module.funcsec.emplace_back(TypeIdx{0});
-    module.globalsec.emplace_back(Global{true, {ConstantExpression::Kind::Constant, {42}}});
-    module.globalsec.emplace_back(Global{true, {ConstantExpression::Kind::Constant, {43}}});
-    module.codesec.emplace_back(Code{0, {Instr::global_get, Instr::end}, {0, 0, 0, 0}});
-    module.codesec.emplace_back(Code{0, {Instr::global_get, Instr::end}, {1, 0, 0, 0}});
-
-    auto instance = instantiate(module);
-
-    auto [trap, ret] = execute(instance, 0, {});
-
-    ASSERT_FALSE(trap);
-    ASSERT_EQ(ret.size(), 1);
-    EXPECT_EQ(ret[0], 42);
-
-    auto [trap2, ret2] = execute(instance, 1, {});
+    /* wat2wasm
+    (global i64 (i64.const 42))
+    (global i64 (i64.const 43))
+    (func (result i64)
+      get_global 0
+    )
+    (func (result i64)
+      get_global 1
+    )
+    */
+    const auto wasm = from_hex(
+        "0061736d010000000105016000017e0303020000060b027e00422a0b7e00422b0b0a0b02040023000b04002301"
+        "0b");
 
-    ASSERT_FALSE(trap2);
-    ASSERT_EQ(ret2.size(), 1);
-    EXPECT_EQ(ret2[0], 43);
+    auto instance = instantiate(parse(wasm));
+    EXPECT_RESULT(execute(instance, 0, {}), 42);
+    EXPECT_RESULT(execute(instance, 1, {}), 43);
 }
 
 TEST(execute, global_get_imported)
 {
-    Module module;
-    module.funcsec.emplace_back(TypeIdx{0});
-    module.importsec.emplace_back(Import{"mod", "glob", ExternalKind::Global, {false}});
-    module.codesec.emplace_back(Code{0, {Instr::global_get, Instr::end}, {0, 0, 0, 0}});
+    /* wat2wasm
+    (import "mod" "glob" (global i64))
+    (func (result i64)
+      get_global 0
+    )
+    */
+    const auto wasm = from_hex(
+        "0061736d010000000105016000017e020d01036d6f6404676c6f62037e00030201000a0601040023000b");
+    const auto module = parse(wasm);
 
-    uint64_t global_value = 42;
+    uint64_t global_value = 0;
```

Comment:
Why not keep this test case?

You could also add `EXPECT_RESULT(execute(), 0);` if you change this.

Reply:
Restored original case and added one more with resetting global value to 0.